src/x/sync/index_cpu.go

```diff
@@ -68,6 +79,11 @@ func CPUCore() int {
 		// and got 1 back.
 		return 0
 	}
+
+	if runtime.GOOS == "darwin" {
```

Comment:
Actually, this isn't quite right. We should be returning the current core here. Will investigate into how we get this.